### ISSUE TITLE
Kernel crashing with an unknow reason

### ISSUE THREAD

**Issue by francescobodria** [tensorflow/tcav#91]
I am trying to use tcav with my CNN keras model, but my kernel keeps crashing for an unknown reason when launching tcav.run().
After checking the code seems that the error is presenting when calling compute_tcav_score in this line of tcav.py     

i_up = self.compute_tcav_score(
        mymodel, target_class_for_compute_tcav_score, cav_concept,
        cav_instance, acts[target_class][cav_instance.bottleneck],
        activation_generator.get_examples_for_concept(target_class),
        run_parallel=run_parallel)

The wrapper I have created for my keras model is the following:

class KerasModelWrapper(model.ImageModelWrapper):
    """ ModelWrapper for keras models

        By default, assumes that your model contains one input node, one output head
    and one loss function.
    Computes gradients of the output layer in respect to a CAV.

    Args:
        sess: Tensorflow session we will use for TCAV.
        model_path: Path to your model.h5 file, containing a saved trained
          model.
        labels_path: Path to a file containing the labels for your problem. It
          requires a .txt file, where every line contains a label for your
          model. You want to make sure that the order of labels in this file
          matches with the logits layers for your model, such that file[i] ==
          model_logits[i]
    """

    def __init__(
        self,
        sess,
        model_path,
        labels_path,
        image_shape
    ):
        self.sess = sess
        self.model_name = 'cifar_CNN'
        super(KerasModelWrapper, self).__init__(image_shape)
        self.import_keras_model(model_path)
        self.labels = tf.io.gfile.GFile(labels_path).read().splitlines()
    
        # Construct gradient ops. Defaults to using the model's output layer
        self.y_input = tf.compat.v1.placeholder(tf.int64, shape=[None])
        self.loss = self.model.loss_functions[0](self.y_input,
                                                 self.model.outputs[0])
        self._make_gradient_tensors()
  
    def id_to_label(self, idx):
        return self.labels[idx]
  
    def label_to_id(self, label):
        return self.labels.index(label)
  
    def import_keras_model(self, saved_path):
        """Loads keras model, fetching bottlenecks, inputs and outputs."""
        self.ends = {}
        self.model = tf.keras.models.load_model(saved_path)
        self.get_bottleneck_tensors()
        self.get_inputs_and_outputs_and_ends()
  
    def get_bottleneck_tensors(self):
        self.bottlenecks_tensors = {}
        layers = self.model.layers
        for layer in layers:
            if 'input' not in layer.name:
                self.bottlenecks_tensors[layer.name] = layer.output
  
    def get_inputs_and_outputs_and_ends(self):
        self.ends['input'] = self.model.inputs[0]
        self.ends['prediction'] = self.model.outputs[0]

I don't know if can help but from the log I can retrieve only this information before the crash
F ./tensorflow/core/util/bcast.h:111] Check failed: vec.size() == NDIMS (1 vs. 2)

What am I doing wrong?

**Comment by jameswex** [comment]
@francescobodria Would you be able to point us to the model files as well so we can test this ourselves and debug? Also point us to the data directories for the images you are using. Thanks!

@a-brain-known-as-ralph FYI

**Comment by francescobodria** [comment]
Sure,
I've zipped all in this file. Since I am using [CIFAR](https://www.cs.toronto.edu/~kriz/cifar.html) dataset the images are very small. The model is called cnn_simple_cifar, the target folder is deer, the concepts folders are soil, grass and garden. 
The model specs are:
Model: "sequential_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
conv2d_1 (Conv2D)            (None, 32, 32, 32)        896       
_________________________________________________________________
conv2d_2 (Conv2D)            (None, 32, 32, 32)        9248      
_________________________________________________________________
max_pooling2d_1 (MaxPooling2 (None, 16, 16, 32)        0         
_________________________________________________________________
dropout_1 (Dropout)          (None, 16, 16, 32)        0         
_________________________________________________________________
conv2d_3 (Conv2D)            (None, 16, 16, 64)        18496     
_________________________________________________________________
conv2d_4 (Conv2D)            (None, 16, 16, 64)        36928     
_________________________________________________________________
max_pooling2d_2 (MaxPooling2 (None, 8, 8, 64)          0         
_________________________________________________________________
dropout_2 (Dropout)          (None, 8, 8, 64)          0         
_________________________________________________________________
conv2d_5 (Conv2D)            (None, 8, 8, 128)         73856     
_________________________________________________________________
conv2d_6 (Conv2D)            (None, 8, 8, 128)         147584    
_________________________________________________________________
max_pooling2d_3 (MaxPooling2 (None, 4, 4, 128)         0         
_________________________________________________________________
dropout_3 (Dropout)          (None, 4, 4, 128)         0         
_________________________________________________________________
flatten_1 (Flatten)          (None, 2048)              0         
_________________________________________________________________
dense_1 (Dense)              (None, 128)               262272    
_________________________________________________________________
dropout_4 (Dropout)          (None, 128)               0         
_________________________________________________________________
dense_2 (Dense)              (None, 10)                1290      
=================================================================
Total params: 550,570
Trainable params: 550,570
Non-trainable params: 0
As bottlenecks, I was trying to use 'conv2d_6'
I can point you to the same google drive folder if you prefer.
[tcav-cifar.zip](https://github.com/tensorflow/tcav/files/5486558/tcav-cifar.zip)

**Comment by OrangeeJi** [comment]
Hi! @francescobodria I am not sure if you have already solved this problem. 
I also met this problem a month ago, and I found my problem was caused by the loss part. 
I used categorical_crossentropy but the label input for gradient calculation is not one-hot representation (it is just a index). So it failed as Check failed: vec.size() == NDIMS (1 vs. 2).
Then I retrained my model with SparseCategoricalCrossentropy loss then it solved the problem.
I am not sure if you are facing the same situation as me.
Just a little hint. Hope this can help you.
